src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migrator.ts

```diff
@@ -12,11 +12,71 @@ import {IMPORT_REPLACEMENTS} from './import-replacements';
 export class RuntimeMigrator {
   oldImportModule: string;
   newImportModule: string;
+  importSpecifierReplacements: {old: string; new: string}[];
 
   constructor(component: string) {
     const replacements = IMPORT_REPLACEMENTS[component];
     this.oldImportModule = replacements.old;
     this.newImportModule = replacements.new;
+
+    const firstLetterCapitalizedComponent = component[0].toUpperCase() + component.slice(1);
+    const capitalizedComponent = component.toUpperCase();
+    this.importSpecifierReplacements = [
+      {
+        old: 'MatLegacy' + firstLetterCapitalizedComponent,
+        new: 'Mat' + firstLetterCapitalizedComponent,
+      },
+      {
+        old: 'MAT_LEGACY_' + capitalizedComponent,
+        new: 'MAT_' + capitalizedComponent,
+      },
+    ];
+  }
+
+  updateImportSpecifier(importSpecifier: ts.Identifier): ts.Identifier | null {
+    const newImport = this._getNewImportSpecifier(importSpecifier);
+    if (!newImport) {
+      // Did not find any specifier that should be replaced
+      return null;
+    }
+
+    return ts.factory.createIdentifier(newImport);
+  }
+
+  updateImportSpecifierWithPossibleAlias(
+    importSpecifier: ts.ImportSpecifier,
+  ): ts.ImportSpecifier | null {
+    // If the import specifier has an alias, there is a property name. For the
+    // example 'MatLegacyButtonModule as MatButtonModule', the property name is
+    // MatLegacyButtonModule and the name is MatButtonModule. Although for an
+    // example with no alias, 'MatLegacyButtonModule', the name is
+    // MatLegacyButtonModule and there is no property name. Since we are
+    // updating the value with the legacy prefix, we have to handle that it can
+    // be in either the name or property name field for the node.
+    const newImport = this._getNewImportSpecifier(
+      importSpecifier.propertyName ?? importSpecifier.name,
+    );
+
+    if (!newImport) {
+      // Did not find any specifier that should be replaced
+      return null;
+    }
+
+    let newPropertyName;
+    let newName;
+    // If this import specifier has an alias, only handle it specially if the
+    // alias isn't the new import since we can remove the alias now. In the
+    // example of 'MatLegacyButtonModule as MatButtonModule', it would become
+    // 'MatButtonModule'.
+    if (importSpecifier.propertyName && importSpecifier.name.getText() !== newImport) {
```

Comment:
Little suggestion. Generally avoid `getText` or `getFullText` as this exists for all nodes but doesn't guarantee the proper static analyzable values. If we know e.g. for import clauses that the specifier names are always `Identifier`'s - we can directly take the identifier-specific `.text` property (which 100% matches the identifier name), while `getText()` could match other unexpected nodes or additional whitespace (in the worst case)

```suggestion
    if (importSpecifier.propertyName && importSpecifier.name.text !== newImport) {
```